app/main.js

```diff
@@ -155,14 +158,63 @@ function startProcessWin () {
   })
   processWin.loadURL(modalPath)
   processWin.once('ready-to-show', () => {
-    planVersionCheck()
+    planVersionCheck()  
   })
 }
 
+function checkForAutoUpdate() {
+  const autoUpdatePromise = autoUpdater.checkForUpdates()
+  if(autoUpdatePromise) {
+    const modalPathUpdateWindow = `file://${__dirname}/stretchlyAutoUpdater.html#v${app.getVersion()}`
+    updaterWindow = new BrowserWindow({
+        icon: `${__dirname}/images/stretchly_18x18.png`,
+        x: displaysX(),
+        y: displaysY(),
+        backgroundColor: settings.get('mainColor'),
+        resizable: false,
+    })
+    updaterWindow.loadURL(modalPathUpdateWindow)
+  }
+}
+const sendStatusToUpdaterWindow = (text) => {
+  if(updaterWindow) {
+    updaterWindow.send('message', text);
+  }
+}
+
 function planVersionCheck (seconds = 1) {
   setTimeout(checkVersion, seconds * 1000)
 }
 
+autoUpdater.on('checking-for-update', () => {
+  sendStatusToUpdaterWindow('Checking for update')
+})
+
+autoUpdater.on('update-available', (info) => {
+  sendStatusToUpdaterWindow('Update Available')
+})
+
+autoUpdater.on('update-not-available', (info) => {
+  sendStatusToUpdaterWindow('Update Not Available')
+})
+
+autoUpdater.on('error', (err) => {
+  sendStatusToUpdaterWindow('Error'+ err)
+})
+autoUpdater.on('download-progress', (progressObj) => {
+      let log_message = "Download speed: " + progressObj.bytesPerSecond;
```

Comment:
This will need to be translatable, but let's deal with it once the functionality is done.